Use of unresolved identifier 'SO_TIMESTAMP'

I'm attempting to recompile a Swift package on my Raspberry Pi 4 originally built with swift-nio-ssl 2.10.0 and now I'm getting multiple errors along the lines of:

.../.build/checkouts/swift-nio/Sources/NIO/BSDSocketAPI.swift:404:43: error: use of unresolved identifier 'SO_TIMESTAMP'
            NIOBSDSocket.Option(rawValue: SO_TIMESTAMP)


### SwiftNIO version/commit hash
swift-nio-ssl.git at 2.13.1

### Swift & OS version (output of `swift --version && uname -a`)
Swift version 5.1.5 (swift-5.1.5-RELEASE)
Target: armv6-unknown-linux-gnueabihf
Linux pi4 5.10.17-v7l+ #1414 SMP Fri Apr 30 13:20:47 BST 2021 armv7l GNU/Linux

I should add that there's also a matching error for a 'SO_RCVTIMEO' constant:

.../.build/checkouts/swift-nio/Sources/NIO/BSDSocketAPI.swift:381:43: error: use of unresolved identifier 'SO_RCVTIMEO'
            NIOBSDSocket.Option(rawValue: SO_RCVTIMEO)

--- a/Sources/CNIOLinux/include/CNIOLinux.h
+++ b/Sources/CNIOLinux/include/CNIOLinux.h
@@ -72,5 +72,9 @@ const void *CNIOLinux_CMSG_DATA(const struct cmsghdr *);
 void *CNIOLinux_CMSG_DATA_MUTABLE(struct cmsghdr *);
 size_t CNIOLinux_CMSG_LEN(size_t);
 size_t CNIOLinux_CMSG_SPACE(size_t);
+
+// awkward time_T pain
+extern int CNIOLinux_SO_TIMESTAMP;
+extern int CNIOLinux_SO_RCVTIMEO;
 #endif
 #endif

--- a/Sources/CNIOLinux/shim.c
+++ b/Sources/CNIOLinux/shim.c
@@ -143,4 +143,7 @@ size_t CNIOLinux_CMSG_LEN(size_t payloadSizeBytes) {
 size_t CNIOLinux_CMSG_SPACE(size_t payloadSizeBytes) {
     return CMSG_SPACE(payloadSizeBytes);
 }
+
+int CNIOLinux_SO_TIMESTAMP = SO_TIMESTAMP;
+int CNIOLinux_SO_RCVTIMEO = SO_RCVTIMEO;
 #endif

--- a/Sources/NIO/System.swift
+++ b/Sources/NIO/System.swift
@@ -58,6 +58,12 @@ let S_IFBLK = UInt32(SwiftGlibc.S_IFBLK)
 #endif
 #endif
 
+// Work around SO_TIMESTAMP/SO_RCVTIMEO being awkwardly defined in glibc.
+#if os(Linux)
+let SO_TIMESTAMP = CNIOLinux_SO_TIMESTAMP
+let SO_RCVTIMEO = CNIOLinux_SO_RCVTIMEO
+#endif
+
 // Declare aliases to share more code and not need to repeat #if #else blocks
 #if !os(Windows)
 private let sysClose = close

